src/Sarif.Viewer.VisualStudio/CodeFlowToTreeConverter.cs

```diff
@@ -1,64 +1,54 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections.Generic;
+using System.Linq;
 using Microsoft.CodeAnalysis.Sarif;
 using Microsoft.Sarif.Viewer.Models;
-using System.Collections.Generic;
 
 namespace Microsoft.Sarif.Viewer.VisualStudio
 {
     internal static class CodeFlowToTreeConverter
     {
         internal static List<CallTreeNode> Convert(CodeFlow codeFlow)
         {
-            int currentCodeFlowIndex = -1;
+            var root = new CallTreeNode { Children = new List<CallTreeNode>() };
+            ThreadFlow threadFlow = codeFlow.ThreadFlows?[0];
 
-            return GetChildren(codeFlow, ref currentCodeFlowIndex, null);
-        }
-
-        private static List<CallTreeNode> GetChildren(CodeFlow codeFlow, ref int currentCodeFlowIndex, CallTreeNode parent)
-        {
-            currentCodeFlowIndex++;
-            List<CallTreeNode> children = new List<CallTreeNode>();
-            bool foundCallReturn = false;
-
-            while (currentCodeFlowIndex < codeFlow.Locations.Count && !foundCallReturn)
+            if (threadFlow != null)
             {
-                switch (codeFlow.Locations[currentCodeFlowIndex].Kind)
+                int lastNestingLevel = 0;
+                CallTreeNode lastParent = root;
+
+                foreach (CodeFlowLocation location in threadFlow.Locations)
                 {
-                    case CodeFlowLocationKind.Call:
-                        var newNode = new CallTreeNode
-                        {
-                            Location = codeFlow.Locations[currentCodeFlowIndex],
-                            Parent = parent
-                        };
-                        newNode.Children = GetChildren(codeFlow, ref currentCodeFlowIndex, newNode);
-                        children.Add(newNode);
-                        break;
+                    var newNode = new CallTreeNode
+                    {
+                        Location = location,
+                        Children = new List<CallTreeNode>()
+                    };
 
-                    case CodeFlowLocationKind.CallReturn:
-                        children.Add(new CallTreeNode
-                        {
-                            Location = codeFlow.Locations[currentCodeFlowIndex],
-                            Children = new List<CallTreeNode>(),
-                            Parent = parent
-                        });
-                        foundCallReturn = true;
-                        break;
+                    if (location.NestingLevel > lastNestingLevel)
```

Comment:
Ok, so you accept multiple-level indents. So you will treat the sequence of nesting levels `[ 0, 4, 1, 5, 2 ]` the same as `[ 0, 1, 0, 1, 0 ]`. I'll write myself a note to ask Michael whether to constrain the sequence in the spec, and if not, what behavior we want.